### ISSUE TITLE
Possibility to use custom slots in form kit schema

### ISSUE THREAD

**Issue by dominikklein** [formkit/formkit#190]
### Describe the feature

At the moment it's not possible to use custom slots inside of a FormKitSchema. Here is only the "default" slot supported.
It would be great when the support will be added because it's also possible with the usage in the template, so i think this should be aligned.

**Comment by justin-schroeder** [comment]
This is definitely something we need to do. Do you have a suggested API for the schema node?

```js
{
  $cmp: 'MyComponent',
  children: [
    // -- this is the default slot already --
  ],
  slots: {
    myNamedSlot: [
      {
        $el: 'h1',
        children: 'hello world'
      }
    ]
  }
}
```

Something like that perhaps?

**Comment by dominikklein** [comment]
Yes, this looks good from my side.

**Comment by justin-schroeder** [comment]
@AustinGrey yeah I think something like that is about right. basically we just need a place where we can define what "variable" receives the scope data. We already do this (variable "assignment") in one other place and thats the `for` loop:

```js
{
  $el: 'li',
  for: ['item', '$items']
}
```

Which is intended to read (in the developers mind) "for item in items" or at least as  best as possible in JSON format that is :) 

Another API proposed by @GustavoFenilli is:

```js


```js
{
  $cmp: 'MyComponent',
  children: [
    {
      $slot: 'scopedSlotName',
      children: ['Hello world']
    }
  ]
```

We could pretty easily add scoped data to that as a top level key — although I do think it is a bit clearer to have "slots" as a top level feature of components nodes as in the first API. However, I'm not totally sold on using the semicolon `scopedSlot:scopedData` — would be nice if the syntax was a bit closer to the assignment in for `for` loop.

**Comment by GustavoFenilli** [comment]
I like the idea of passing possible scoped variables as items inside the array and appending the slot name to it so it is indeed scoped and easy to find.

Having the slots at top level wouldn't matter for vue, as slots are passed to the dom correctly using the `<slot>` inside the component, but what about React/Svelte and others that do not have this feature, what order and where would you show thoses slots? if its inside children, we know the order and where.

Slots are basically in concept just a function with scoped variables that renders to the dom, where its rendered is the trick part, but we could also just the top level named slots defined to render inside the children just like `$slots.default`.

```js
{
  $cmp: 'MyComponent',
  slots: [
    { 
      name: 'userDefined',
      scope: ['var1', 'var2'],
      schema: []
    }
  ],
  children: [
    '$slots.userDefined'
  ]
```
